storybookjs/storybook
# FIX ref getSourceType for paths with */index.html

Issue: #10324 

* Issue is described in detail at https://github.com/storybookjs/storybook/issues/10324#issuecomment-613295255 - in short, if path ends with `index.html`, storybook breaks. I (along with anyone doing this at my company) can't fix this by removing `index.html` from the browser url because an internal platform requires it in order to render the file.

## What I did
* Changes `getSourceType` to return `'local'` if source matches location with `index.html` in pathname.
* Added `refs.test.js` to test this edge case and other cases - *I needed to mock different values with the `global.location`. Not too happy with the implementation but I don't know another way to change a mock module between tests.*


## How to test
* Unit tests included in PR
* Reproduce:
  * Follow CONTRIBUTING.md steps and run `yarn serve-storybooks`
  * Navigate to: http://127.0.0.1:8001/official-storybook/index.html
  * See the loading icon does not disappear, "No preview" message visible and no stories listed in nav. Open console for error message.
* Fix:
  * Same steps as above, open http://127.0.0.1:8001/official-storybook/index.html
  * No errors, storybook rendered as expected.
